app/scripts/background.js

```diff
@@ -199,7 +200,13 @@ browser.runtime.onConnectExternal.addListener(async (...args) => {
   // Queue up connection attempts here, waiting until after initialization
   await isInitialized;
   // This is set in `setupController`, which is called as part of initialization
-  connectExternal(...args);
+  const port = args[0];
+
+  if (port.sender.tab?.id && process.env.BARAD_DUR) {
+    connectExternalCaip(...args);
+  } else {
+    connectExternalExtension(...args);
```

Comment:
i think we need the legacy API in all cases

Reply:
To be clear, there are 3 APIs now. Legacy, Current/EIP1193, CAIP

When you say Legacy, I assume you are referring to the Current/EIP1193.

I don't believe we are serving EIP1193 over externally_connectable connections from webpages. @adonesky1

Reply:
yeah. @shanejonas just to clarify, this `browser.runtime.onConnectExternal.addListener(...` block only for externally_connectable connections and we don't need to serve the `EIP1193` provider to any consumers over that transport